File: leetcode/217.containsDuplicate.js
nums = [1, 1, 1, 3, 3, 4, 3, 2, 4, 2];
nums = [3, 3];
var containsDuplicate = function (nums) {
    //
    // sorting method
    //
    // nums.sort();
    // for (let i = 0; i < nums.length - 1; i++) {
    //     if (nums[i] == nums[i + 1]) {
    //         return true;
    //     }
    // }

    //
    // hashmap method
    //

    let hashmap = new Map();

    for (let i = 0; i < nums.length; i++) {
        if (hashmap.has(nums[i])) {
            return true;
        }
        hashmap.set(nums[i], true);
    }

    return false;
};

console.log(containsDuplicate(nums));

File: leetcode/easy/217.containsDuplicate.py
# https://leetcode.com/problems/contains-duplicate/

nums = [1,2,3,4]

def containsDuplicate(nums) -> bool:
    
    # # sorting method
    # nums.sort()
    # for i in range(len(nums)-1):
    #     if nums[i] == nums[i+1]:
    #         return True

    # hashmap method
    hashM = {}
    for i in range(len(nums)-1):
        # check for existing value in hashmap
        if nums[i] in hashM and hashM[nums[i]] == hashM[nums[i+1]]:
            return True
        # else add elements with the value true
        hashM[nums[i]] = True

    return False


print(containsDuplicate(nums))
